lightly/data/collate.py

```diff
@@ -476,4 +478,187 @@ def __init__(self,
             crop_min_scales=crop_min_scales,
             crop_max_scales=crop_max_scales,
             transforms=transforms,
-        )
\ No newline at end of file
+        )
+
+class MultiViewCollateFunction(nn.Module):
+    """Generates multiple views for each image in the batch.
+
+    Attributes:
+        transforms:
+            List of transformation functions. Each function is used to generate
+            one view of the back.
+    
+    """
+    def __init__(self, transforms: List[torchvision.transforms.Compose]):
+        super().__init__()
+        self.transforms = transforms
+
+    def forward(self, batch: List[tuple]):
+        """Turns a batch of tuples into a tuple of batches.
+
+        Args:
+            batch:
+                The input batch.
+        
+        Returns:
+            A (views, labels, fnames) tuple where views is a list of tensors
+            with each tensor containing one view of the batch.
+
+        """
+        views = []
+        for transform in self.transforms:
+            view = torch.stack([transform(img) for img, _, _ in batch])
+            views.append(view)
+        # list of labels
+        labels = torch.LongTensor([label for _, label, _ in batch])
+        # list of filenames
+        fnames = [fname for _, _, fname in batch]
+        return views, labels, fnames
+
+
+class DINOCollateFunction(MultiViewCollateFunction):
```

Comment:
What's the reason you're not inheriting from the `MultiCropCollateFunction`?

Reply:
Because dino uses different transforms for some of the views. For example the first global view has no solarization while the second one has solarization. This is not supported by the `MultiCropCollateFunction` and I did not want to refactor it and make breaking changes.

Reply:
I see...

Reply:
I think it would make sense to refactor the SwAV collate function to use the MultiView collate function in the future as it is more generic and supports any number of views and transforms.

Reply:
As a first step we could make the `MultiCropCollate` function inherit from the `MultiViewCollate`. Then the concepts are related at least.